.travis.yml

```diff
@@ -32,7 +32,13 @@ jobs:
       name: "integration test edge"     # names the fourth Tests stage job
     - script: make edge_cross_build
       name: "cross build edge"     # names the fifth Tests stage job
-    - script: make e2e_test
+    - script:
+      | if ! git diff --name-only $TRAVIS_COMMIT_RANGE | grep -qvE '(.md)'
```

Comment:
What will happen if a commit includes two changes: one for `foo.go` and the other one for `bar.md`?

Reply:
grep is quiet, invert-match, and extend-regexp mode. 
I think invert-match means if match except (.md) then it returns true.
And list of filenames reads OR conditions that even only one NOT .md file returns true. 
e.g.
grep -qvE foo.md is false
grep -qvE foo1.md is false
grep -qvE bar.go is true
OR conditions of filelist is true
So, invert-if statement pass through the else conditions that executes e2e test.
c.f. https://reflectoring.io/skip-ci-build/

Reply:
Okay, thanks for clarification. 

Per the blog, why not

 `if ! git diff --name-only $TRAVIS_COMMIT_RANGE | grep -qvE '(.md$)'`?

Reply:
sorry, typo found.`$`